packages/react/src/components/radio-button-group/radio-button-group.tsx

```diff
@@ -127,7 +127,9 @@ export const RadioButtonGroup: VoidFunctionComponent<RadioButtonGroupProps> = ({
     ...otherProps
 }) => {
     const { isMobile } = useDeviceContext();
-    const [currentChecked, setCurrentChecked] = useState(checkedValue);
+    const [currentChecked, setCurrentChecked] = useState(
+        buttons.find((button) => button.defaultChecked)?.value || checkedValue,
```

Comment:
The checkedValue is the controlled version, it should take precedence over the default.  defaultChecked should only be used if there is no specified checkedValue.